### ISSUE TITLE
missing global mobile data killswitch causing roaming data charges

### ISSUE THREAD

**Issue by twojstaryzdomu** [osmandapp/OsmAnd#15389]
### Description
Making this bug specific to a key feature of apps that feature regular updates. Users have no ablility to suspend all updates with one toggle in an easily accessible location. The missing switch is essential to anyone frequently crossing borders, since map updates on a roaming network are likely to lead to excessive charges. A single killswitch for suppressing updates while on mobile data is necessary.
The discrete switches hidden several levels in the menu are not a viable alternative. With several maps installed, the user is forced to toggle the maps individually. For border crossers, it is a chore. Disabling background data is not an option either since updates would run anyway when the map is in the foreground. The global data killswitch like in most other apps needs to be in the app itself.

Please note, this bug is not catch-all for all map update bugs or issues and should be considered a separate bug at its face value due to the financial impact on all border crossing users, normally subject to high roaming network charges for data use while abroad. A necessity for frequent travellers.

### How to reproduce?
Non-existent option

### Your Environment
OsmAnd Version: 4.1.11
Android/iOS version: Android
Device model: any

**Maps used (online or offline):**   
Concerns all offline maps

**Comment by tradavyvy** [comment]
This looks like a repeat entry of closed issue #15365

**Comment by vshcherb** [comment]
To avoid such impact internet should be disabled every app nowadays could download smth in background and cause significant impact. I personally always limit speed if I'm low on internet and it helps to realize that something downloads  I don't see any invisible OsmAnd place that could cause massive download

**Comment by twojstaryzdomu** [comment]
> To avoid such impact internet should be disabled every app nowadays could download smth in background and cause significant impact.

I believe this question has already been answered: while the background killswitch will be fine for most other apps, map apps tend to be on. Users typically need data abroad for lower volume apps like email or IM so it isn't a feasible solution to throw out the baby with the bath water and disable all data altogether for the sake of a single app.
A single toggle that you can flick and disable all map updates in map would be ideal here. Imagine a user realising their maps are downloading large data volume while driving abroad. Those minuscule update switches per map will have to be flipped one by one, which can be a chore for many maps. Absolutely not a thing to do while driving.

**Comment by tradavyvy** [comment]
I don't think you should do ANY setup while driving!

I still do not see why you should not set all maps to no automatic update and just manualy update when wifi is available (quite often nowadays)
Exept maybe when you are an avid mapper and want to see all your changes immediately. And even then just set your activeapping area to automatic update.

**Comment by EssBee59** [comment]
An other solution under Android to restrict the data usage of the Osmand app!
(long press on the icon, ==> app info ==> restrict data usage ==> set restriction for Mobile data, but not for Wi-Fi!)

Using this setup, you can avoid all dowloads from the app on mobile data.
(an error message is displayed when Osmand tries a download on mobile data)

**Comment by twojstaryzdomu** [comment]
> An other solution under Android is to "restrict the data usage" of the Osmand app!
> (long press on the icon, ==> app info ==> restrict data usage ==> set restriction for Mobile data, but not for Wi-Fi!)

Thank you I do recall such an option from LineageOS but I'm afraid it may not be available on stock that comes with the phone, at least on an fairly popular Android 12 phone by a large manufacturer.

**Comment by sonora** [comment]
It's a standard Android feature, both in 11 and 12.

**Comment by twojstaryzdomu** [comment]
> It's a standard Android feature, both in 11 and 12.

Is it? No such toggle I'm afraid on a very popular vendor's Android 12.  Background data usage can be toggled per app only, optionally limited only to the data saving mode.

**Comment by twojstaryzdomu** [comment]
> driving!

Indeed. The fact remains users do that anyway, regardless of the warning. Pulling over is an option though. In any case, a big handy switch turn off mobile data downloads would be welcome for users who first encounter the problem going abroad, rather than a series of small per map switches.

**Comment by sonora** [comment]
Try this procedure: https://www.androidhelpcenter.com/2021/12/restrict-background-data-usage-on-android/

It will inhibit any mobile (not WiFi) data usage for any selected app. And for foreground usage, i.e. while you display OsmAnd, I am not aware OsmAnd uses any data for anything you do not explicitly trigger, so no issue there either.

**Comment by twojstaryzdomu** [comment]
@sonora have you read my statement above? The procedure you gave is known to me a

> It will inhibit any mobile (not WiFi) data usage

> Only the background data usage can be toggled per app, optionally limited only to the data saving mode. No mobile data per app toggle.

This issue is about the not being a toggle 1)  in the app 2) for mobile data only, NOT background data only. Foreground mobile data needs to be covered by the toggle as well as background.

It IS an issue to users that will find OSMand downloading their updates abroad. Please try to get into their shoes: you normally run into this kind of an issue unbeknownst to you, since map updates are enabled by default. Then, user looks for a switch to disable mobile data in the app and finds none. Do you expect users to keep disabling all the small switches per each map every time they abroad? It's an unreasonable expectation.

That's the reason to have such a SINGLE toggle in an easily accessible location in the app. It's not like users are experts like developers and you need to consider such cases.

Huge bills that they will inevitably run while abroad might cause them to abandon the app. Not a good thing, is it?

**Comment by sonora** [comment]
Maybe I am missing the forest for the trees here... just to clarify: When you say 

> map updates are enabled by default.

That confuses me: All maps are only updated when you explicitly (manually) trigger an update. And in case  you are talking about OsmAnd Live: That already has an overall master switch, located at the top of its download page?

**Comment by twojstaryzdomu** [comment]
> That already has an overall master switch, located at the top of its download 

That may well stay enabled. The bug spells mobile data killswitch, not an overall data killswitch. I hope the difference is clear. When on wifi abroad, updates are fine. It's for the sake of tollable data connections over the mobile network.

**Comment by tradavyvy** [comment]
In my humble opinion this 'problem' has enough possible solutions and or workarounds, and I think the programmers might spend their time more usefull on other (pressing) problems.
I motion to close this subject.

**Comment by twojstaryzdomu** [comment]
> possible solutions and or workarounds

Please name one that:
1) keeps updates going on wifi
2) disables all map updates with a single switch, rather than per map
3) isn't buried several levels in the menu so that it may be easily toggled back & forth

In my view the solution requies one switch that toggles all update on wifi settings for each map. Those are already in place. Now need a global one.

**Comment by tradavyvy** [comment]
Just set all the mentioned sugestions to not update anny maps automaticly. And activate updates manualy whenever you have wifi and you feel you need updated maps.
-Wifi is available verry often.
-Maps do not change verry significant from minute to minute (or even day to day).
-'Update all maps' is not burried verry deep in the mennu structure.
-It is realy not that complicated or hard to do.

**Comment by twojstaryzdomu** [comment]
Toggling updates on and off requies a conscious effort every time. That is not what this bug is about.

This bug is about the lack of a set & forget global mobile data killswitch or wifi only toggle that simultaneously turns all map updates on and off on.


Nothing less nothing more. Why it's only availble per map individually and not globally is beyond my understanding.

**Comment by pebogufi** [comment]
> ... is **beyond my understanding**

So - the main problem of this topic is your understanding ?

**Comment by twojstaryzdomu** [comment]
> the main problem of this topic is your understanding ?

Please restate the question, it doesn't make sense in English.

**Comment by tradavyvy** [comment]
Somebody, PLEASE close this issue.
Thanks

**Comment by twojstaryzdomu** [comment]
And why would that be? Is this a deliberate attempt to be provocative? 

I believe the bug affects anyone who travels abroad and wants to have updates over wifi and locally in their home country over mobile as well.

**Comment by pebogufi** [comment]
> ... it doesn't make sense ...

So it matches to the comments of the initial writer

**Comment by twojstaryzdomu** [comment]
> So it matches to some comments of the initial writer

Would you mind speaking more clearly in English and quoting the context as a matter of intellectual integrity? Thank you.

**Comment by pebogufi** [comment]
In this forum "not native speaker" and "not native thinker" meet each other :-)
I am one of them.

**Comment by pebogufi** [comment]
No, it is not a discussion about any bug, but a discussion of your displeasure.

**Comment by twojstaryzdomu** [comment]
> No, it is not a discussion about any bug

I doubt a person making offensive provocative comments or wordplay to derail a meritocratic discussion can be trusted with their judgement.

**Comment by pebogufi** [comment]
> ... a person ...  can be trusted with their judgement.
After rereading your comments I confirm that.

**Comment by twojstaryzdomu** [comment]
Thanks. If you can make meritocratic comments, by all means do so, if not please stop.
